core/src/main/java/software/amazon/awssdk/retry/RetryPolicyContext.java

```diff
@@ -121,6 +131,11 @@ public Builder exception(SdkBaseException exception) {
             return this;
         }
 
+        public Builder executionContext(ExecutionContext executionContext) {
```

Comment:
`ExecutionContext` feels like a big, internal thing to expose to a public interface like the retry policy. Would it be better to expose the `ExecutionAttributes`, which is what we currently expose to the interceptors for the same reason?

Reply:
+1, sorry that was my suggestion and got the names mixed up.